### PR TITLE
BIOWDL-521: Update tasks to a uniform layout/structure.

### PR DESCRIPTION
**Pull request by JasperBoom**
### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] `parameter_meta` for each task is up to date.

This is mostly just a layout/structure update for the tasks.
I have mainly reordered the parameter_meta to match the order of the inputs.
Separated the inputs into three sections: required/defined inputs, optional
inputs and runtime inputs (which was already the case in a lot of the newer
tasks, but not in the older ones).
I have added new lines between sections within a task, or removed unnecessary
newlines.
And I have added some changes to tasks that I also put in the CHANGELOG.

Don't worry, I did not spend all my time on this, I also did useful things :worried:.

### CODE REVIEW

File: bam2fastx.wdl [hunk 1]
```diff
@@ -38,6 +38,21 @@ task Bam2Fasta {
...
+        do
+            ln $bamFile .
+            bamFiles=$bamFiles" $(basename $bamFile)"
+        done
+
+        for index in ~{sep=" " bamIndex}
+        do
+            ln $index .
+        done
+
```
Comment (DavyCats) [inline bam2fastx.wdl:55]: Isn't this repeated from below?

File: bwa-mem2.wdl [hunk 2]
```diff
@@ -26,33 +26,35 @@ task Mem {
...
-        Int? memoryGb 
+
+        String? readgroup
+        Int? sortThreads
+        Int? memoryGb
+
+        Int threads = 4
         Int timeMinutes = 1 + ceil(size([read1, read2], "G") * 220 / threads)
         # Contains bwa-mem2 2.0 bwakit 0.7.17.dev1 and samtools 1.10
         String dockerImage = "quay.io/biocontainers/mulled-v2-6a15c99309c82b345497d24489bee67bbb76c2f6:1c9c3227b9bf825a8dc9726a25701aa23c0b1f12-0"
```
Comment (DavyCats) [inline bwa-mem2.wdl:41]:
```suggestion

        String? readgroup
        Int? sortThreads
        
        Int? memoryGb
        Int threads = 4
        Int timeMinutes = 1 + ceil(size([read1, read2], "G") * 220 / threads)
        # Contains bwa-mem2 2.0 bwakit 0.7.17.dev1 and samtools 1.10
        String dockerImage = "quay.io/biocontainers/mulled-v2-6a15c99309c82b345497d24489bee67bbb76c2f6:1c9c3227b9bf825a8dc9726a25701aa23c0b1f12-0"
```

File: bwa.wdl [hunk 3]
```diff
@@ -26,28 +26,30 @@ task Mem {
...
-        Int? sortThreads
         Int sortMemoryPerThreadGb = 2
         Int compressionLevel = 1
-        Int? memoryGb 
+
+        String? readgroup
+        Int? sortThreads
+        Int? memoryGb
+
+        Int threads = 4
```
Comment (DavyCats) [inline bwa.wdl:38]:
```suggestion

        String? readgroup
        Int? sortThreads
        
        Int? memoryGb
        Int threads = 4
```

File: bam2fastx.wdl (hunk 1 above)
Comment (JasperBoom) [reply bam2fastx.wdl:55]: It is indeed the same as below. This new entry is part of the `Bam2Fasta` task. The below is `Bam2Fastq`.
Previously, only the `Bam2Fastq` task had this localization. But the `Bam2Fasta` requires it as well.

Comment (DavyCats) [reply bam2fastx.wdl:55]: Oh it's a different task XD

**Review by DavyCats — APPROVED** [review]

File: common.wdl [hunk 4]
```diff
@@ -116,8 +116,8 @@ task Copy {
 }
```
Comment (cagaser) [inline common.wdl:116]: Missing parameters meta, also for the others in this file

File: cutadapt.wdl [hunk 5]
```diff
@@ -152,8 +156,8 @@ task Cutadapt {
 
     output{
```
Comment (cagaser) [inline cutadapt.wdl:157]: missing outputs parameter_meta

File: deepvariant.wdl [hunk 6]
```diff
@@ -59,36 +59,36 @@ task RunDeepVariant {
...
         File outputVCF = outputVcf
         File outputVCFIndex = outputVCF + ".tbi"
+        Array[File] outputVCFStatsReport = glob("*.visual_report.html")
         File? outputGVCF = outputGVcf
         File? outputGVCFIndex = outputGVcf + ".tbi"
-        Array[File] outputVCFStatsReport = glob("*.visual_report.html")
     }
-    
+
     parameter_meta {
```
Comment (cagaser) [inline deepvariant.wdl:75]: Output parameter_meta missing

**Review by cagaser — APPROVED** [review]
Seperate PR should be made to add the missing parameter meta entries.

**Pull request merged**